sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/configuration/model/matching/Detection.java

```diff
@@ -20,27 +20,47 @@
 
 package org.sonar.plugins.secrets.configuration.model.matching;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import javax.annotation.Nullable;
+import org.sonar.plugins.secrets.configuration.model.Rule;
 import org.sonar.plugins.secrets.configuration.model.matching.filter.PostModule;
 import org.sonar.plugins.secrets.configuration.model.matching.filter.PreModule;
 
 public class Detection {
+
+  @Nullable
   private Match matching;
   @Nullable
   private PreModule pre;
   @Nullable
   private PostModule post;
+  @JsonIgnore
+  private Rule rule;
+
+  private boolean associatedProviderDetectionExists() {
+    if (rule != null && rule.getProvider() != null) {
+      return rule.getProvider().getDetection() != null;
+    }
```

Comment:
IMO you do not need to check the provider to be null since a rule always should have a provider.